src/App.js

```diff
@@ -328,7 +300,24 @@ function App() {
 }
 
 function Root() {
-  return (
+  const [initializing, setInitializing] = React.useState(true);
+
+  React.useEffect(() => {
+    async function initializeApplication() {
+      await initSentry(); // must be set up immediately
+
+      const deviceId = await getDeviceId();
+
+      Sentry.setUser({ id: deviceId });
+      // Segment.identify(deviceId)
+
+      setInitializing(false);
+    }
+
+    initializeApplication();
+  }, [setInitializing]);
```

Comment:
Basically, we're just showing the splash screen for a few ms longer while we load this up front.